code/processes/wdb.q

```diff
@@ -203,27 +203,32 @@ endofdaysave:{[dir;pt]
 	.lg.o[`savefinish;"finished saving data to disk"];
 	};
 
-/- add entries to dictionary of callbacks. if timeout has expired or d now contains all expected rows then it releases each waiting process
+/- add entries to keyed table of callbacks. if timeout has expired or d now contains all expected rows then it releases each waiting process
 handler:{
-	.wdb.d[.z.w]:x;
-	if[(.proc.cp[]>.wdb.timeouttime) or (count[.wdb.d]=.wdb.countreload);
-		.lg.o[`handler;"releasing processes"];
-		.wdb.flushend[];
-		.wdb.d:()!()];
+	 if[not .z.w in  key .wdb.d;
+        .wdb.d[.z.w]:x];
+	if[(.proc.cp[]>.wdb.timeouttime) or (.wdb.countreload=count .wdb.d);
+        	.lg.o[`reload;string[count select from .wdb.d where status=1]," out of ", string[count .wdb.d]," processes successfully reloaded"];
+        	.lg.o[`handler;"releasing processes"];
+		.wdb.flushend[]];
+		if[.wdb.reloadcomplete;.wdb.d:([handle:()]process:();status:();result:())];
```

Comment:
probably better to do a "delete from" as it means if the structure of the table were to change you only have to change it in one place

Reply:
Have edited this to a delete. However, would this be an issue in the scenario where the last process to reload fails? As the .wdb.handler is called twice re:
```
  [({@[`. `reload;x;
        /-if process fails to reload message with error entered into .wdb.d and  output error to process error log
                                {[ptype;e](neg .z.w)(`.wdb.handler;(ptype;0b;`$"failed with error: ",e));
                                .lg.e[`reloadproc;"failed to reload ",string[ptype]," from .wdb.reloadproc call. The error was : ",e]}[y]];
        /-Successful reload message to be sent to handler and entered into .wdb.d for end of reload summary logging
                                 (neg .z.w)(`.wdb.handler;(y;1b;`$"successfully reloaded")); (neg .z.w)[]};d;ptype);h;ptype];
```
As the error trap message will cause the reload end process to run i.e. calling .wdb.flushend[] and setting .wdb.reloadcomplete to 1b and deleting .wdb.d. The second .wdb.handler call will execute while .wdb.d has been deleted. Which I think will an output print to the err_wdb1.log file as the initial if statements in .wdb.handler reference .wdb.d.